packages/valory/skills/termination_abci/tests/test_behaviours.py

```diff
@@ -0,0 +1,547 @@
+    ]
+    _NUM_ZERO_TRANSFER_EVENTS = len(_ZERO_TRANSFER_EVENTS)
+    # when the service owner got removed from being a safe owner
+    # this happens when the operators take over the safe again
+    _SERVICE_OWNER_REMOVED_EVENTS = [
+        {"block_number": 2},
+        {"block_number": 11},
+    ]
+    _NUM_SERVICE_OWNER_REMOVED_EVENTS = len(_SERVICE_OWNER_REMOVED_EVENTS)
+    _SAFE_OWNERS = ["0x1", "0x2", "0x3", "0x4"]
+    _NUM_SAFE_OWNERS = len(_SAFE_OWNERS)
+    _SAFE_THRESHOLD = 1
+    _MOCK_TX_RESPONSE = b"0xIrrelevantForTests".hex()
+    _MOCK_TX_HASH = "0x" + "0" * 64
+
+    _STATE_ERR_LOG = (
+        f"Expected response performative {ContractApiMessage.Performative.STATE.value}, "  # type: ignore
+        f"received {ContractApiMessage.Performative.ERROR}."
+    )
+    _RAW_TRANSACTION_ERR = (
+        f"Expected response performative {ContractApiMessage.Performative.RAW_TRANSACTION.value}, "  # type: ignore
+        f"received {ContractApiMessage.Performative.ERROR}."
+    )
+    _SERVICE_OWNER_ERR_LOG = (
+        f"Couldn't get the service owner for service with id={SERVICE_ID}. "
+        f"{_STATE_ERR_LOG}"
+    )
+    _ZERO_TRANSFER_EVENTS_ERR_LOG = (
+        f"Couldn't get the latest Zero Transfer (`SafeReceived`) event. "
+        f"{_STATE_ERR_LOG}"
+    )
+    _SERVICE_OWNER_REMOVED_EVENTS_ERR_LOG = (
+        f"Couldn't get the latest `RemovedOwner` event. " f"{_STATE_ERR_LOG}"
+    )
+    _SAFE_OWNERS_ERR_LOG = (
+        f"Couldn't get the safe owners for safe deployed at {SAFE_ADDRESS}. "
+        f"{_STATE_ERR_LOG}"
+    )
+    _REMOVE_OWNER_ERR_LOG = f"Couldn't get a remove owner tx. " f"{_STATE_ERR_LOG}"
+    _SWAP_OWNER_ERR_LOG = f"Couldn't get a swap owner tx. " f"{_STATE_ERR_LOG}"
+    _SAFE_HASH_ERR_LOG = f"Couldn't get safe hash. " f"{_STATE_ERR_LOG}"
+    _MULTISEND_ERR_LOG = "Couldn't compile the multisend tx. "
+    _SUCCESS_LOG = "Successfully prepared termination multisend tx."
+
+    def _mock_get_service_owner_request(
+        self,
+        error: bool = False,
+    ) -> None:
+        """Mock a ServiceRegistryContract.get_service_owner() request."""
+        if not error:
+            response_performative = ContractApiMessage.Performative.STATE
+            response_body = dict(service_owner=SERVICE_OWNER_ADDRESS)
+        else:
+            response_body = dict()
+            response_performative = ContractApiMessage.Performative.ERROR
+
+        self.mock_contract_api_request(
+            contract_id=str(ServiceRegistryContract.contract_id),
+            request_kwargs=dict(
+                performative=ContractApiMessage.Performative.GET_STATE,
+                contract_address=SERVICE_REGISTRY_ADDRESS,
+            ),
+            response_kwargs=dict(
+                performative=response_performative,
+                state=State(
+                    ledger_id="ethereum",
+                    body=response_body,
+                ),
+            ),
+        )
```

Comment:
these mock contract api calls are the only thing that is difficult to read and understand. Do a CTRL + F on `mock_contract_api_request` and skim over them hitting enter, you'll find 8 nearly identical code blocks. I suggest to introduce a helper function that only takes the variable arguments.